### DISCUSSION TITLE
Where to find the birt runtime?

### DISCUSSION THREAD

**Discussion by colinsutton** [eclipse-birt/birt#796] [Q&A] [answered]
Following the excellent video instructions I have managed to build the BIRT all-in-one for Windows with yesterday's changes from Wim on my local clone of birt-master. It runs my reports using an Oracle database quite happily, no longer any problem with the viewer. I would also like to test my issue (https://github.com/eclipse/birt/issues/693) with genreport on my local system, but I can't figure out the process required to build a local copy. What do I select and where?

**Comment by hvbtup** [comment, marked as answer]
When you built BIRT from inside Eclipse - fine.
You can generate the runtime with Maven. In my experience the easiest way to do this is from inside Eclipse, too.

![grafik](https://user-images.githubusercontent.com/7644776/151328798-fcbddd9d-dbdc-452c-8d05-edcb14f87f39.png)

For the top-level-project in the Project Explorer, choose "Run As... / Maven build..." from the context menu.

In the Goals field, enter: clean package
![grafik](https://user-images.githubusercontent.com/7644776/151329147-1fdaf659-0dd6-4577-ae21-b04a6c8cb162.png)

Then click the "Run" button and wait...
...
...
...
...
That takes a very long time on my machine...

Finally, you should see "Build successful" in the console.

Then, use the file explorer to navigate to git\birt\build\birt-packages\birt-runtime\target.
You should find a new file birt-runtime-4.9.0-{date}.zip here. That's your home-brewed BIRT runtime. It includes ReportEngine\lib.

PS
Greetings to my dentist P Sherman, 42 Wallaby Way, Sydney :-)

**Reply by colinsutton** [reply]
> When you built BIRT from inside Eclipse - fine. You can generate the runtime with Maven. In my experience the easiest way to do this is from inside Eclipse, too.
Tried, but the build failed today (on windows):
<img width="1288" alt="screenshot_581" src="https://user-images.githubusercontent.com/18058891/151487218-44348583-4c8a-485c-bf0e-77d64c1407c4.png">

P.S.
I have not seen Nemo

**Reply by hvbtup** [reply]
That error message cannot be from the Maven script.
However, I had build errors yesterday, too.
The error messages looked a bit different, but I think the reason is similar.

In my case the error message from the Maven build looked like this:

Missing requirement: ... requires 'org.eclipse.equinox.p2.iu; ... but it could not be found
I don't remember the exact content of the ....

I don't really understand that OSGI mess, but what I did was: 

First I opened the .target file, clicked the "Reload Target Platform" in the top-right corner.
That didn't help.
Next I tried the "Reload" button 
![grafik](https://user-images.githubusercontent.com/7644776/151506376-3cb4e2b4-a196-4770-b1fb-14b4c6f04f1a.png)
on the same page.
That still didn't help.

I remembered from earlier that it once didn't work due to something with OSGI not included.
I opened the run configuration and saw this:
![grafik](https://user-images.githubusercontent.com/7644776/151513924-a7008d7e-ad0d-4141-90de-85dacb3a5128.png)
Two different version of org.eclipse.osgi, but none of them was selected.
So I selected the newer one.

This still didn't help, but at least it resulted in a different error message, this time the message from the Maven build looked like this:

Missing requirement: org.eclipse.birt.report.designer.all 4.9.0.qualifier requires 'org.eclipse.equinox.p2.iu; org.eclipse.ecf.filetransfer.httpclient45.feature.feature.group 0.0.0' but it could not be found

Then I called a younger colleague with more Eclipse/OSGI experience.
We finally found that in the .product file, httpclient (the very last entry) showed httpclient4.5 with a little red error marker.
So we removed this and added httpclient5 instead:
![grafik](https://user-images.githubusercontent.com/7644776/151514560-e78ea9b4-28e9-42a3-9eb7-84c35190a270.png)

Only then did the build succeed.

**Reply by colinsutton** [reply]
> > Tried, but the build failed today (on windows):

I deleted the 'Fragment host' line from the manifest and rebuilt. This time I had a successful build . Now I feel I'm getting somewhere.

**Reply by wimjongman** [reply]
The build instructions outside Eclipse are in the README.MD in the root of the project.

**Comment by wimjongman** [comment]
You can also download a pre-built version from the saved artifacts in the Actions.

**Reply by colinsutton** [reply]
> You can also download a pre-built version from the saved artifacts in the Actions.

Yes, that version works OK on Windows if I edit the genReport.bat file. But I'd like to be able to build and test my changes to the bat file and, eventually, understand how to build for different platforms.
